### PR TITLE
Add private network posture check

### PR DESCRIPTION
**Pull request by bcmmbaga**
### Checklist
- [ ] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: management/server/http/api/openapi.yml [hunk 1]
```diff
@@ -932,6 +934,24 @@ components:
...
+    PrivateNetworkCheck:
+      description: Posture check for allow or deny private network
+      type: object
+      properties:
+        prefixes:
+          description: List of private network prefixes
+          type: array
+          items:
+            type: string
+            example: "192.168.1.0/24"
```
Comment (mlsmaycon) [inline management/server/http/api/openapi.yml]:
```suggestion
            example: ["192.168.1.0/24", "10.0.0.0/8"]
```

File: management/server/http/api/openapi.yml [hunk 2]
```diff
@@ -932,6 +934,24 @@ components:
       required:
         - locations
         - action
+    PrivateNetworkCheck:
+      description: Posture check for allow or deny private network
+      type: object
+      properties:
+        prefixes:
```
Comment (mlsmaycon) [inline management/server/http/api/openapi.yml]: Let's call them ranges

File: management/server/http/api/openapi.yml [hunk 3]
```diff
@@ -932,6 +934,24 @@ components:
...
+          description: List of private network prefixes
```
Comment (mlsmaycon) [inline management/server/http/api/openapi.yml]:
```suggestion
          description: List of private network ranges in CIDR notation
```

File: management/server/http/posture_checks_handler.go [hunk 4]
```diff
@@ -342,3 +368,31 @@ func toPostureGeoLocationCheck(apiGeoLocationCheck *api.GeoLocationCheck) *postu
...
+	}
+
+	return &api.PrivateNetworkCheck{
+		Prefixes: netPrefixes,
+		Action:   api.PrivateNetworkCheckAction(check.Action),
+	}
+}
+
+func toPrivateNetworkCheck(check *api.PrivateNetworkCheck) (*posture.PrivateNetworkCheck, error) {
+	prefixes := make([]netip.Prefix, 0)
```
Comment (mlsmaycon) [inline management/server/http/posture_checks_handler.go:385]: we should check if the len of incoming ranges > 0 and return an error otherwise

**Review by mlsmaycon — CHANGES_REQUESTED** [review]

File: management/server/http/posture_checks_handler.go (hunk 4 above)
Comment (bcmmbaga) [reply management/server/http/posture_checks_handler.go:385]: This is already checked during [validation](https://github.com/netbirdio/netbird/blob/2ba3f38db7ac782bcfcee9cd6acb27c848b0d429/management/server/http/posture_checks_handler.go#L290).

**Review by mlsmaycon — APPROVED** [review]

**Pull request merged**